presto-spi/src/main/java/com/facebook/presto/spi/session/SessionPropertyRule.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.session;
+
+import java.util.Map;
+
+public interface SessionPropertyRule
+{
+    Map<String, String> match(SessionConfigurationContext context);
```

Comment:
We shouldn't expose a dotted string to connectors. The catalog / property name split should be modeled explicitly, similar to how we have `SchemaTableName` rather than just a string.